### ISSUE TITLE
commit 8644e9d broke .imn deployment

### ISSUE THREAD

**Issue by gsomlo** [coreemu/core#162]
While .imn files still load and look fine in core-gui, when the simulation is started the actual resulting node configuration is badly broken.

For an example, compare the intended bgp configuration for "router1" in /usr/share/core/examples/configs/sample3-bgp.imn with the output of "show run" from vtysh.

I bisected from back around 5.0 when things still worked, and the culprit seems to be commit 8644e9d, "moved future core server and handler code to act as the default core-daemon ..."

Unfortunately that's didn't help me isolate the actual problem, since it's swapping out one big chunk of code for another -- so it looks like I need professional help figuring out what went wrong :)

Thanks,
--Gabriel

**Comment by bharnden** [comment]
I did some quick poking around without being a quagga expert. ifconfig and ip addr show the right addresses for the scenario. This file also overrides the default zebra service and excludes some information for some vtysh configuration, that would normally be provided.

```bash
service integrated-vtysh-config
````

That didn't seem to change anything that appeared from the vtysh command itself. router1 can ping the other local nodes. Not sure if the wrong information is isolated to whats displayed by the command or something else is going on and there is a way to verify.

**Comment by gsomlo** [comment]
On Thu, May 24, 2018 at 12:14:25PM -0700, bharnden wrote:
> I did some quick poking around without being a quagga expert. ifconfig and ip
> addr show the right addresses for the scenario. This file also overrides the
> default zebra service and excludes some information for some vtysh
> configuration, that would normally be provided.
> 
> service integrated-vtysh-config
> 
> That didn't seem to change anything that appeared from the vtysh command
> itself. router1 can ping the other local nodes. Not sure if the wrong
> information is isolated to whats displayed by the command or something else is
> going on and there is a way to verify.

My suspicion is that there's a bug in how the daemon parses/interprets
the configuration being handed over from the core-gui client. It's not
just quagga, other containers with e.g. user-defined service and data
files end up completely wrong in my other simulations. Just a hunch,
though, no actual evidence...

On the example in question (sample3-bgp.imn, node "router1", the
content of /etc/quagga/Quagga.conf on the deployed container is
different from what's specified in the .imn file, so I'm guessing it
gets corrupted during the above-mentioned parsing/handover from client
to daemon. Shouldn't be a quagga/bgp specific problem, though...

**Comment by bharnden** [comment]
There's definitely something we saw going on with custom service files and when I checked it appeared as if it may always have been an issue, bay I may be wrong and need to take a deeper look.

If I take the sample3 file and remove the custom-config sections and let the service use their defaults, which seem to be the same anyway, then things look better.

I'll play around with comparing previous versions and see where it went off, if it was broken.
